### ISSUE TITLE
Providing a map version of `dataUri`

### ISSUE THREAD

**Issue by thom4parisot** [bbc/peaks.js#44]
# tl;dr

The `dataUri` is pretty weak at the moment because of the feature detection.  
It also implies URLs to end with `.dat` or `.json`.
# How it works
## Simplified version

We can specify what is the format associated to that URL.

``` js
peaks.init({
  dataUri: 'http://server/whatever/you/want',
  defaultDataFormat: 'arraybuffer'
});
```

Or provide a map of URLs:

``` js
peaks.init({
  dataUri: {
    arraybuffer: '/path/to/file/my/binary/file',
    json: '/path/to/file/json'
  }
});
```
## Fallback version

More complex but handles possible custom formats.

They iterate in the declared order and if `configure` returns a falsy value or throws an `Error`, it fallbacks to the next available format.

``` js
peaks.init({
  dataUri: [
    {
      format: 'arraybuffer',
      uri: '/path/to/file/my/binary/file',
      configure: function (xhr) {
        xhr.responseType = 'arraybuffer';
        return xhr;
      },
    {
      format: 'json',
      uri: '/path/to/file/json',
      configure: function (xhr) {
        try {  xhr.responseType = 'json'; }
        catch (e){
          console.log('xhr.responseType is not supported, the response handler will try a JSON parse before failing.');
        }

        return xhr;
      }
  ]
});
```
## Using provided factories

Same as above but it hides the complexity.

``` js
peaks.init({
  dataUri: [
    peaks.data.arraybuffer('/path/to/file/arraybuffer.bin'),
    peaks.data.json('/path/to/file/json')
  ]
});
```

Works well with a single format. It assumes the `defaultDataFormat` from the `factory.format` property.

``` js
peaks.init({
  dataUri: peaks.data.arraybuffer('/path/to/file/arraybuffer.bin')
});
```
